crates/orderbook/src/database/orders.rs

```diff
@@ -356,6 +359,68 @@ impl OrderStoring for Postgres {
         .try_collect()
         .await
     }
+
+    async fn order_status(&self, order_uid: &OrderUid) -> Result<Status> {
+        let mut ex = self.pool.begin().await.context("could not init tx")?;
+        let timer = super::Metrics::get()
+            .database_queries
+            .with_label_values(&["latest_order_event"])
+            .start_timer();
+        let status = database::order_events::get_latest(&mut ex, &ByteArray(order_uid.0))
+            .await
+            .context("could not fetch status")?;
+        timer.stop_and_record();
+
+        let fetch_solvers = || async move {
+            let timer = super::Metrics::get()
+                .database_queries
+                .with_label_values(&["load_latest_solver_competition"])
+                .start_timer();
+            let competition = database::solver_competition::load_latest_competition(&mut ex)
+                .await
+                .context("could not fetch latest competition")?
+                .unwrap()
+                .json;
+            timer.stop_and_record();
+
+            let competition: SolverCompetitionDB = serde_json::from_value(competition)
+                .context("could not parse solver competition data")?;
+
+            let solvers = competition
+                .solutions
+                .into_iter()
+                .filter_map(|solution| {
+                    solution
+                        .orders
+                        .iter()
+                        .any(|o| match o {
+                            model::solver_competition::Order::Colocated { id, .. } => {
+                                id.0 == order_uid.0
+                            }
+                            model::solver_competition::Order::Legacy { id, .. } => {
+                                id.0 == order_uid.0
+                            }
+                        })
+                        .then_some(solution.solver)
+                })
+                .collect::<Vec<_>>();
+
+            Ok::<Vec<_>, anyhow::Error>(solvers)
+        };
+
+        let status = match status.label {
```

Comment:
This could be done by implementing the `From<>` trait

Reply:
`From` interface accepts `T` only, while we also need here a lazy list of solvers.